### ISSUE TITLE
Upstream into `tokio-compat`?

### ISSUE THREAD

**Issue by hawkw** [Nemo157/futures-tokio-compat#2]
Hi @Nemo157,

I'm currently [working on a `tokio-compat` crate](https://github.com/tokio-rs/tokio/pull/1663) in the Tokio repo. Currently, the `tokio-compat` crate added in tokio-rs/tokio#1663 adds modified versions of tokio's runtimes that can provide a `tokio` 0.1 reactor and timer, and support spawning both `std::future` and `futures` 0.1 tasks. However, another goal for the `tokio-compat` crate is to provide `AsyncRead`/`AsyncWrite` compatibility as well (see tokio-rs/tokio#1549) until compatibility issues between `tokio`'s versions of those traits and `futures-io`'s versions are resolved.

Since you've already written this code, I was wondering if you would be interested in contributing it to the upstream `tokio-compat` crate? If you're interested, the way I see it, there are two ways we could go about this:
- once tokio-rs/tokio#1663 lands, you could open a PR to add the `AsyncRead`/`AsyncWrite` compat code to the `tokio-compat` crate (I'd probably place it in a `io` module and add a default-on feature flag). 
- alternatively, if you're interested in upstreaming this code but don't have time to open a PR, I could open a PR adding this code. I could add attribution to you in the commit that merges your code in.

If you'd prefer not to upstream this crate into `tokio-compat`, that's also fine, and we can come up with a separate implementation — I just thought that since you've already done the work, it was worth reaching out.

**Comment by Nemo157** [comment]
I'd be very happy for this to be upstreamed. I won't have time to open a PR soon, so feel free to take whatever helps you get it working.

It's really very straightfoward code since I avoided any of the complex APIs, as it's so trivial I don't really care about attribution at all.

**Comment by hawkw** [comment]
Great, thank you. I'll provide some attribution regardless :)
